app/api/jobs/channels/route.ts

```diff
@@ -1,35 +1,63 @@
+import distributeChannelStakerWeeklyAirdropAndTips from "@/lib/sonata/channelWeeklyAirdrop/distributeChannelStakerWeeklyAirdropAndTips";
+import distributeChannelWeeklyAirdrop from "@/lib/sonata/channelWeeklyAirdrop/distributeChannelWeeklyAirdrop";
 import sortChannels from "@/lib/sortChannels";
 import { stack } from "@/lib/stack/client";
+import { eventTipChannel } from "@/lib/stack/events";
+import getDaysChannelTotalTips from "@/lib/stack/getDaysChannelTotalTips";
 import getChannelStats from "@/lib/supabase/getChannelStats";
+import supabase from "@/lib/supabase/serverClient";
 import { NextRequest } from "next/server";
 
 export async function GET(req: NextRequest) {
   const TOP_CHANNELS = 10;
-  const TIP_AMOUNT = 11111;
+  const AIRDROP_AMOUNT = 11111;
 
-  const authHeader = req.headers.get('authorization');
-  if (authHeader !== `Bearer ${process.env.CRON_SECRET}`) {
-    return Response.json({ message: 'Unauthorized' }, { status: 401 });
+  if (process.env.NEXT_PUBLIC_VERCEL_ENV === 'production') {
+    const authHeader = req.headers.get('authorization');
+    if (authHeader !== `Bearer ${process.env.CRON_SECRET}`) {
+      return Response.json({ message: 'Unauthorized' }, { status: 401 });
+    }
   }
 
-  const channelStats = await getChannelStats(true);
-  const sortedChannels = sortChannels(channelStats);
-  sortedChannels.splice(TOP_CHANNELS);
-
-  const stacks = await Promise.all(sortedChannels.map((channel: any) => {
-    const address = channel?.addresses?.[0];
-    if (!address)
-      return Promise.reject(`${channel.channelId} channel address not found`);
-
-    return stack.track(`weekly_channel_tip_to_${address}`, {
-      account: address, points: TIP_AMOUNT
-    });
-  }));
-
-  if (stacks.some(res => !res.success))
-    return Response.json({ message: 'failed', stacks }, { status: 500 });
-
-  return Response.json({ message: 'success', topChannels: sortedChannels });
+  try {
+    const channelStats = await getChannelStats(true);
+    const sortedChannels = sortChannels(channelStats);
+    sortedChannels.splice(TOP_CHANNELS);
+
+    const results = await Promise.all(sortedChannels.map(async (channel) => {
+      const channelId = channel.channelId;
+
+      const balance = channel.staked ? await getDaysChannelTotalTips(channelId) : 0;
```

Comment:
could you please explicitly pass `7` here?
- why? what if a future dev changes the default value?